git/git_filter_protocol.go

```diff
@@ -36,103 +38,75 @@ func NewObjectScanner(r io.Reader, w io.Writer) *ObjectScanner {
 	}
 }
 
-func (o *ObjectScanner) Init() bool {
+func (o *ObjectScanner) Init() error {
 	tracerx.Printf("Initialize filter")
 	reqVer := "version=2"
 
 	initMsg, err := o.p.readPacketText()
 	if err != nil {
-		fmt.Fprintf(os.Stderr,
-			"Error: reading filter initialization failed with %s\n", err)
-		return false
+		return fmt.Errorf("reading filter initialization failed with %s", err)
```

Comment:
Ooh, definitely agree. This was moving some old code around, and I didn't even think to add that. Got it in d484908fb8418dcaedf9a69fb3d1c6087345c6af.